CONTRIBUTING.md

```diff
@@ -0,0 +1,115 @@
+
+Most discussions happen on our [Discord Server](https://discord.zelda64.dev), where you are welcome to ask if you need help getting started, or if you have any questions regarding this project and other decompilation projects.
+
+
+Useful Links
+------------
+
+- [Zelda 64 Reverse Engineering Website](https://zelda64.dev/) - Our homepage, with links to other resources.
+- [Installation guide](https://github.com/zeldaret/mm/wiki/Installation) - Instructions for getting this repository set up and built on your machine.
+- [Introduction to OOT decomp](https://github.com/zeldaret/oot/blob/master/docs/tutorial/contents.md) - A very detailed tutorial on how to get started with decomp. For OOT, but largely applicable to MM as well.
+
+- [Discord](https://discord.zelda64.dev/) - Primary discussion platform.
+- [Trello board](https://trello.com/b/ruxw9n6m/majoras-mask-decompilation) - We use this to track decompilation progress, not GitHub Issues.
+
+Getting Started
+---------------
+
+### What should I know to take part?
+
+Basic knowledge of C, particularly arrays and pointers, is extremely useful. Knowledge of MIPS is not required initially, you can pick it up as you work through some decompilation.
+
+You should be familiar with using git and GitHub. There are a number of tutorials available online, [such as this one](https://github.com/firstcontributions/first-contributions), which can help you get started.
+
+The most useful knowledge to have is a general understanding of how the game works. An afternoon of constructive mucking about in [OOT's Practice Rom](http://practicerom.com/) (aka GZ) will be very beneficial if you have not looked at either of the N64 Zelda's internals before.
+
+This project only uses *publicly available code*. In particular, do not use any Nintendo source code leaks.
+
+### Environment Setup
+
+Get started by following our [installation guide](https://github.com/zeldaret/mm/wiki/Installation).
+When successful, you should be able to build a matching ROM before you start making any changes.
+
+### First Contribution
+
+Usually, the best place to get started is to decompile an actor overlay. 
+An *actor* is any thing in the game that moves or performs actions or interactions: Link is an actor, enemies are actors, NPCs are actors, etc.
+
+You can [join the Discord](https://discord.zelda64.dev/) to say hello and get suggestions on where to start on the `#mm-decomp` channel.
+We track who is working on what on the [Trello board](https://trello.com/b/ruxw9n6m/majoras-mask-decompilation).
+
+Style Guide & Conventions
+-------------------------
+
+Most of the C formatting style is enforced by the `format.sh` script, which is based on `clang-format`.
+Running `./format.sh` will apply our standard style to all `.c` files in the repository.
+
+There are some conventions that cannot be automatically enforced.
+
+(TODO)
+- Naming scheme
+- `ActorFunc` naming specifics
+- Renaming symbols referenced by asm (?)
+- Check for new warnings
```

Comment:
Could we add that struct and enum definitions for actors should go in the actor header files?
Also making sure that the actor data in the file uses the `static` keyword. 
Just two ideas off the top of my head